app/src/main/java/co/touchlab/kampkit/android/MainActivity.kt

```diff
@@ -1,101 +1,50 @@
 package co.touchlab.kampkit.android
 
 import android.os.Bundle
-import android.view.View
+import androidx.activity.compose.setContent
 import androidx.appcompat.app.AppCompatActivity
-import androidx.lifecycle.lifecycleScope
-import androidx.recyclerview.widget.LinearLayoutManager
-import androidx.recyclerview.widget.RecyclerView
-import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
-import co.touchlab.kampkit.android.adapter.MainAdapter
-import co.touchlab.kampkit.android.databinding.ActivityMainBinding
-import co.touchlab.kampkit.db.Breed
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import co.touchlab.kampkit.android.ui.MainScreen
+import co.touchlab.kampkit.android.ui.theme.KaMPKitTheme
 import co.touchlab.kampkit.models.DataState
 import co.touchlab.kermit.Kermit
-import com.google.android.material.snackbar.Snackbar
-import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.launch
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import kotlinx.coroutines.flow.map
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 import org.koin.core.parameter.parametersOf
 
 class MainActivity : AppCompatActivity(), KoinComponent {
```

Comment:
I've also seen most Compose samples using `ComponentActivity`. `AppCompatActivity` works fine, but I'd like to understand why `ComponentActivity` is more popular in Compose. https://developer.android.com/jetpack/compose/interop/interop-apis#compose-in-views
> If you want to incorporate Compose UI content in a fragment or an existing View layout, use ComposeView and call its setContent() method. ComposeView is an Android View. You must attach the ComposeView to a ViewTreeLifecycleOwner. The ViewTreeLifecycleOwner allows the view to be attached and detached repeatedly while preserving the composition. ComponentActivity, FragmentActivity and AppCompatActivity are all examples of classes that implement ViewTreeLifecycleOwner.